contrib/gofiber/fiber.v2/option.go

```diff
@@ -76,12 +77,8 @@ func WithAnalyticsRate(rate float64) Option {
 }
 
 // WithStatusCheck allow setting of a function to tell whether a status code is an error
-func WithStatusCheck(fn func(statusCode int) bool) Option {
+func WithStatusCheck(b bool) Option {
```

Comment:
This is a publicly-exposed API, and we can't change it's signature.
Instead, why don't we:
1. mark this as deprecated and point to the new global status codes.
2. use the WithStatusCheck function if it's set, else use the global status codes.

Reply:
Originally, we had a private function `isServerError` which handled the provided status code and checked if it was a server error. Would it be better to handle checking if a status code is in the global bitset in this private function so that we don't have to deprecate `WithStatusCheck`?